opentelemetry-sdk/src/opentelemetry/sdk/trace/__init__.py

```diff
@@ -336,6 +340,11 @@ def _filter_attribute_values(attributes: types.Attributes):
                 attributes.pop(attr_key)
 
 
+# pylint: disable=protected-access
+def _make_event_attributes_immutable(event: EventBase) -> None:
```

Comment:
Nit: I don't think you need these helper functions, especially since it is only being called once.